pkg/remotewrite/remotewrite.go

```diff
@@ -138,7 +145,7 @@ func (o *Output) staleMarkers(t time.Time) []*prompb.TimeSeries {
 //
 // TODO: refactor, the code can be improved
 func (o *Output) setTrendStatsResolver(trendStats []string) error {
-	var trendStatsCopy []string
+	var trendStatsCopy []string //nolint:prealloc
```

Comment:
Mmm..can we do something smarter to remove it? :thinking:

Reply:
I tried, but because we essentially are performing a filter operation here, we can't know the size beforehand. An alternative would be to preallocate a slice with length 0, and capacity set to the length of `trendStats`. What way, worst case scenario, no values are filtered out, and the underlying dynamic array has pre-set enough storage.

Thus: 
```go
trendStatsCopy := make([]string, 0, len(trendStats))
```

What do you think?

Reply:
It sounds fine to me